cache-config/testing/ort-tests/t3c-no-outgoing-ip-flag_test.go

```diff
@@ -19,72 +19,49 @@ import (
 	"errors"
 	"io/ioutil"
 	"os/exec"
-	"path/filepath"
 	"strconv"
+	"strings"
 	"testing"
 
 	"github.com/apache/trafficcontrol/cache-config/testing/ort-tests/tcdata"
 	"github.com/apache/trafficcontrol/lib/go-util"
 )
 
+const outgoingIPToBind = "outgoing_ip_to_bind"
+
+func testNoOutgoingIPAfterUpdate(t *testing.T, noOutgoingIP *bool) {
+	if err := t3cUpdateNoOutgoingIP(cacheHostName, noOutgoingIP); err != nil {
+		t.Fatalf("t3c badass failed: %v", err)
+	}
+
+	recordsDotConfig, err := ioutil.ReadFile(recordsConfigFileName)
+	if err != nil {
+		t.Fatalf("reading %s: %v", recordsConfigFileName, err)
+	}
+	contents := string(recordsDotConfig)
+
+	if noOutgoingIP == nil || !*noOutgoingIP {
```

Comment:
The de-duplication is great. But this is some subtle logic embedded in here. Namely that a nil value defaults to false. There's no indication in the larger test or this function name, that this is the logic that's expected to happen.

It's really a coincidence that the input and output are both bools. For any other input type, a table of input-to-value would be ideal. But since it's bool-to-bool, I guess a comment would be fine, versus that level of abstraction. Alternatively, some kind of `const defaultIfNil = false` would also solve the subtlety issue.

Reply:
Given that it's the behavior of the actual utility and not the test to default to "false", I feel a constant would best be introduced in the utility itself, not the tests. So I'm gonna go for the comment.